solr/core/src/java/org/apache/solr/cloud/OverseerTaskQueue.java

```diff
@@ -222,7 +222,7 @@ String createResponseNode() throws KeeperException, InterruptedException {
   }
 
 
-  public List<QueueEvent> peekTopN(int n, Set<String> excludeSet, long waitMillis)
+  public List<QueueEvent> peekTopN(int n, final Set<String> excludeSet, long waitMillis)
```

Comment:
@noblepaul I could go ahead and make this take a Function to better align with your change if you like, assuming you're not going the Sets.union() option